--- a/lisp/cus-edit.el	2026-01-15 22:04:36
+++ b/lisp/cus-edit.el	2026-01-15 22:04:36
@@ -2255,11 +2255,11 @@
 ;;; The `custom' Widget.
 
 (defface custom-button
-  '((((type x w32 ns haiku pgtk android) (class color)
+  '((((type x w32 ns haiku pgtk android ios) (class color)
       (min-colors 88))	; Like default mode line
      :box (:line-width 2 :style released-button)
      :background "lightgrey" :foreground "black")
-    (((type x w32 ns haiku pgtk android))
+    (((type x w32 ns haiku pgtk android ios))
      :box (:line-width 2 :style released-button)
      :background "white" :foreground "black"))
   "Face for custom buffer buttons if `custom-raised-buttons' is non-nil."
@@ -2267,11 +2267,11 @@
   :group 'custom-faces)
 
 (defface custom-button-mouse
-  '((((type x w32 ns haiku pgtk android) (class color)
+  '((((type x w32 ns haiku pgtk android ios) (class color)
       (min-colors 88))
      :box (:line-width 2 :style released-button)
      :background "grey90" :foreground "black")
-    (((type x w32 ns haiku pgtk android))
+    (((type x w32 ns haiku pgtk android ios))
      :box (:line-width 2 :style released-button)
      ;; Either light gray or a stipple pattern.
      :background "gray20" :foreground "black")
@@ -2297,10 +2297,10 @@
       (if custom-raised-buttons 'custom-button-mouse 'highlight))
 
 (defface custom-button-pressed
-  '((((type x w32 ns haiku pgtk android) (class color grayscale))
+  '((((type x w32 ns haiku pgtk android ios) (class color grayscale))
      :box (:line-width 2 :style pressed-button)
      :background "lightgrey" :foreground "black")
-    (((type x w32 ns haiku pgtk android))
+    (((type x w32 ns haiku pgtk android ios))
      :box (:line-width 2 :style pressed-button)
      ;; Either light gray or a stipple pattern.
      :background "gray20" :foreground "black")
