--- a/src/character.c
+++ b/src/character.c
@@ -23,6 +23,10 @@ along with GNU Emacs.  If not, see <https://www.gnu.org/licenses/>.  */
 
 #include <config.h>
 
+#ifdef HAVE_IOS
+#include <ctype.h>
+#endif
+
 #include <stdio.h>
 
 #include "lisp.h"
@@ -955,6 +959,10 @@ wordy (int c)
 bool
 alphabeticp (int c)
 {
+#ifdef HAVE_IOS
+  if (!CHAR_TABLE_P (Vunicode_category_table))
+    return c < 128 && isalpha (c);
+#endif
   Lisp_Object category = CHAR_TABLE_REF (Vunicode_category_table, c);
   if (! FIXNUMP (category))
     return false;
@@ -978,6 +986,10 @@ alphabeticp (int c)
 bool
 alphanumericp (int c)
 {
+#ifdef HAVE_IOS
+  if (!CHAR_TABLE_P (Vunicode_category_table))
+    return c < 128 && isalnum (c);
+#endif
   Lisp_Object category = CHAR_TABLE_REF (Vunicode_category_table, c);
   if (! FIXNUMP (category))
     return false;
@@ -1000,6 +1012,10 @@ alphanumericp (int c)
 bool
 graphicp (int c)
 {
+#ifdef HAVE_IOS
+  if (!CHAR_TABLE_P (Vunicode_category_table))
+    return c < 128 && isgraph (c);
+#endif
   Lisp_Object category = CHAR_TABLE_REF (Vunicode_category_table, c);
   if (! FIXNUMP (category))
     return false;
@@ -1018,6 +1034,10 @@ graphicp (int c)
 bool
 printablep (int c)
 {
+#ifdef HAVE_IOS
+  if (!CHAR_TABLE_P (Vunicode_category_table))
+    return c < 128 && isprint (c);
+#endif
   Lisp_Object category = CHAR_TABLE_REF (Vunicode_category_table, c);
   if (! FIXNUMP (category))
     return false;
@@ -1033,6 +1053,10 @@ printablep (int c)
 bool
 punctuationp (int c)
 {
+#ifdef HAVE_IOS
+  if (!CHAR_TABLE_P (Vunicode_category_table))
+    return c < 128 && ispunct (c);
+#endif
   Lisp_Object category = CHAR_TABLE_REF (Vunicode_category_table, c);
   if (! FIXNUMP (category))
     return false;
@@ -1055,6 +1079,10 @@ punctuationp (int c)
 bool
 blankp (int c)
 {
+#ifdef HAVE_IOS
+  if (!CHAR_TABLE_P (Vunicode_category_table))
+    return c == ' ' || c == '\t';
+#endif
   Lisp_Object category = CHAR_TABLE_REF (Vunicode_category_table, c);
   if (! FIXNUMP (category))
     return false;
